### PR TITLE
fix and renaming moveSamplesheets to moveAndCheckSamplesheets + updated config files

### PR DESCRIPTION
**Pull request by RoanKanninga**

### CODE REVIEW

**Comment by RoanKanninga** [comment]
Done testing. Can be merged

File: bin/copyRawDataToTmp.sh [hunk 1]
```diff
@@ -47,21 +47,45 @@ fi
...
-			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    to ${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/rawdata/ngs/"
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Looping through all prm mounts"
+		copied="no"
+		for prm in "${ALL_PRM[@]}"
+		do
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Checking for ${line} on ${prm} and if it still needs to be processed"
+			if [[ -f "/groups/${group}/${prm}/rawdata/ngs/${line}" && "${copied}" == "no" ]]
+			then	
+				log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${line} found on ${prm}, start rsyncing.."
+				rsync -rlDvc --relative --rsync-path="sudo -u ${group}-ateambot rsync" "/groups/${group}/${prm}/./rawdata/ngs/${line}"* "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/" \
```
Comment (pneerincx) [inline bin/copyRawDataToTmp.sh:63]: Why is `t` to preserve time stamps dropped from the `rsync` command?

File: bin/copyRawDataToTmp.sh [hunk 2]
```diff
@@ -47,21 +47,45 @@ fi
...
-		|| {
-			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Failed to rsync ${line}"
-			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    from ${WORKING_DIR}/./rawdata/ngs/"
-			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    to ${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/rawdata/ngs/"
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Looping through all prm mounts"
+		copied="no"
+		for prm in "${ALL_PRM[@]}"
+		do
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Checking for ${line} on ${prm} and if it still needs to be processed"
+			if [[ -f "/groups/${group}/${prm}/rawdata/ngs/${line}" && "${copied}" == "no" ]]
```
Comment (pneerincx) [inline bin/copyRawDataToTmp.sh:60]: - The `-f` test will only work for files and hence fail if we want to transfer a dir or symlink in the future. Why not test for whether the path exists. E.g. 
   ```
   if [[ -e "/groups/${group}/${prm}/rawdata/ngs/${line}" && "${copied}" == "no" ]]
   ```
- If `${line}` would also contain `rawdata/ngs` then this script could also work for other pipelines like the ones for processing array data. Now `rawdata/ngs/` is hard coded in front of the `${line}`making it only work for one type of data.

File: bin/copyRawDataToTmp.sh [hunk 3]
```diff
@@ -47,21 +47,45 @@ fi
...
+		then
+			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Rsync failed, data is not found! Searched for the following:"
+			for prm in "${ALL_PRM[@]}"
+			do
+				log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "/groups/${group}/${prm}/rawdata/ngs/${line}"
+			done
 			mv "${JOB_CONTROLE_FILE_BASE}."{started,failed}
-			return
-		}
+			exit 1
```
Comment (pneerincx) [inline bin/copyRawDataToTmp.sh:85]: `exit 1` betekent dat het script eruit klapt. Aangezien deze functie wordt aangeroepen in een loop ```for sampleSheet in "${sampleSheetsFolder[@]}"``` worden eventuele volgende samplesheets niet meer verwerkt. Kortom een fout voor één samplesheet, maakt dat de verwerking van de rest stokt. Dat lijkt me niet de bedoeling. Er moet een "*.failed" gemaakt worden om de signaleren via `notifications.sh` dat het mis is gegaan met deze samplesheet, maar daarna moet dit script wel doorgaan en de rest verwerken.

File: bin/moveAndCheckSamplesheets.sh [hunk 4]
```diff
@@ -49,7 +49,7 @@ function showHelp() {
 	#
 	cat <<EOH
 ===============================================================================================================
-Script to move samplesheets to another location potentially on another server.
+	Script to check samplesheets and move to another location potentially on another server.
```
Comment (pneerincx) [inline bin/moveAndCheckSamplesheets.sh:52]:
```suggestion
Script to check samplesheets and move to another location potentially on another server.
```

File: bin/moveAndCheckSamplesheets.sh [hunk 5]
```diff
@@ -192,41 +185,111 @@ samplesheetsSource="${DAT_ROOT_DIR}/samplesheets/new/"
...
 	IFS="${SAMPLESHEET_SEP}" read -r -a _sampleSheetColumnNames <<< "$(head -1 "${samplesheet}")"
 	
 	for (( _offset = 0 ; _offset < ${#_sampleSheetColumnNames[@]} ; _offset++ ))
 	do
 		_sampleSheetColumnOffsets["${_sampleSheetColumnNames[${_offset}]}"]="${_offset}"
 	done
+	
+	if [[ -n "${_sampleSheetColumnOffsets["SentrixBarcode_A"]+isset}" ]] 
+	then
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "This is a GAP samplesheet, there is no samplesheetCheck at this moment"
```
Comment (pneerincx) [inline bin/moveAndCheckSamplesheets.sh:224]:
```suggestion
		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "This is a GAP samplesheet. There is no samplesheetCheck at this moment."
```

File: bin/moveAndCheckSamplesheets.sh [hunk 6]
```diff
@@ -192,41 +185,111 @@ samplesheetsSource="${DAT_ROOT_DIR}/samplesheets/new/"
...
+	else
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "This is a NGS samplesheet, lets check if the samplesheet is correct"
```
Comment (pneerincx) [inline bin/moveAndCheckSamplesheets.sh:226]:
```suggestion
		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "This is a NGS samplesheet. Lets check if the samplesheet is correct."
```

File: bin/moveAndCheckSamplesheets.sh [hunk 7]
```diff
@@ -192,41 +185,111 @@ samplesheetsSource="${DAT_ROOT_DIR}/samplesheets/new/"
...
+		#  3. We remove empty lines.
+		#
+		printf '\n'     >> "${samplesheet}.converted"
+		sed 's/\r/\n/g' "${samplesheet}.converted" > "${samplesheet}.converted.tmp"
+		sed '/^\s*$/d'  "${samplesheet}.converted.tmp" > "${samplesheet}.converted.tmp2"
+		rm "${samplesheet}.converted.tmp"
+		mv "${samplesheet}.converted.tmp2" "${samplesheet}.converted"
+		if checkSampleSheet.py --input "${samplesheet}.converted" --log "${samplesheet}.converted.log"
+		then
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "samplesheet is correct"
```
Comment (pneerincx) [inline bin/moveAndCheckSamplesheets.sh:242]:
```suggestion
			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Samplesheet ${samplesheet}.converted is correct."
```

File: bin/moveAndCheckSamplesheets.sh [hunk 8]
```diff
@@ -192,41 +185,111 @@ samplesheetsSource="${DAT_ROOT_DIR}/samplesheets/new/"
...
+			check=$(cat "${samplesheet}.converted.log")
+		else
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "samplesheet is not correct"
```
Comment (pneerincx) [inline bin/moveAndCheckSamplesheets.sh:245]:
```suggestion
			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Samplesheet ${samplesheet}.converted is not correct."
```

File: bin/moveAndCheckSamplesheets.sh [hunk 9]
```diff
@@ -192,41 +185,111 @@ samplesheetsSource="${DAT_ROOT_DIR}/samplesheets/new/"
...
+			check=$(cat "${samplesheet}.converted.log")
+		fi
+		projectSamplesheet="false"
+		if [[ "${check}" == *'OKAY'* ]]
+		then
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Samplesheet is correct, continue"
```
Comment (pneerincx) [inline bin/moveAndCheckSamplesheets.sh:251]: This looks redundant as you already concluded based on only the exit status of ```if checkSampleSheet.py ...``` whether the samplesheet was correct or not.

File: bin/moveAndCheckSamplesheets.sh [hunk 10]
```diff
@@ -192,41 +185,111 @@ samplesheetsSource="${DAT_ROOT_DIR}/samplesheets/new/"
...
-	#shellcheck disable=SC2153
-	samplesheetsDestination="${HOSTNAME_TMP}:/groups/${GROUP}/${SCR_LFS}/Samplesheets/${firstStepOfPipeline}/new/"
+
+
+	# Check whether the samplesheet is a project samplesheet (no NGS_Demultiplexing)
+	# needs an update when we are going to use VIP into production
+	if [[ ${projectSamplesheet} == "true" ]]
+	then
+		firstStepOfPipeline="NGS_DNA"
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "the samplesheet is a project samplesheet (no NGS_Demultiplexing), samplesheet moved to ${firstStepOfPipeline}"
```
Comment (pneerincx) [inline bin/moveAndCheckSamplesheets.sh:284]:
```suggestion
		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "The samplesheet is a project samplesheet (no NGS_Demultiplexing); firstStepOfPipeline was set to ${firstStepOfPipeline}."
```
Note that I do not this moving of the samplesheet

**Comment by pneerincx** [comment]
General comment for `moveAndCheckSamplesheets.sh`: all logic is in a `main` with variables that have global scope. That can de dangerous and provide unexpected results. Our scripts usually try to keep the main code and list of global variables as short as possible: just a loop over whatever needs to be processed and for each item that needs to be processed a function is called. That function can then use `local` to declare variables which are only valid within the scope of that function and hence only valid for a specific item of the loop. By convention the names of these local variables start with an underscore. Currently there are global variables whose names start with underscores: they were probably copied from a function to the main of this script.

**Review by pneerincx — CHANGES_REQUESTED** [review]
See various comments.

**Review by kdelange — APPROVED** [review]

**Pull request merged**